js/app.js

```diff
@@ -39,9 +39,7 @@ angular.module('PathOfDamage', [])
   };
 
   $scope.updateDamageValues = function (skipSerialization) {
-    for (var i = 0; i < $scope.hits.length - 1; i++) {
-      $scope.hits[i] = calcDamage($scope.hits[i].hit);
-    }
+    $scope.hits = $scope.hits.map(function (hit) { return calcDamage(hit.hit) });
```

Comment:
Does this not throw an exception? The last hit is null because it's a blank line, which is why I iterate from 0 to hits.length - 1 in the previous loop.